### PR TITLE
Add workflow tag

### PR DESCRIPTION
**Pull request by mbenguig**

### CODE REVIEW

File: src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java [hunk 1]
```diff
@@ -190,6 +190,13 @@ public Set<String> listContentTypes() {
         return catalogObjectService.getContentTypes();
     }
 
+    @ApiOperation(value = "Lists all workflow tags for all objects")
```
Comment (fviale) [inline src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java:193]: Lists all tags for all objects

and the request should be /tags

Maybe tags will not only be available for workflow objects

File: src/main/java/org/ow2/proactive/catalog/service/CatalogObjectService.java [hunk 2]
```diff
@@ -482,19 +482,28 @@ public List<CatalogObjectRevisionEntity> listCatalogObjectsEntities(List<String>
...
+                                  contentType,
+                                  Optional.empty(),
+                                  Optional.empty(),
+                                  0,
+                                  Integer.MAX_VALUE);
     }
 
     public List<CatalogObjectMetadata> listCatalogObjects(List<String> bucketsNames, Optional<String> kind,
-            Optional<String> contentType, Optional<String> objectNameFilter, int pageNo, int pageSize) {
+            Optional<String> contentType, Optional<String> objectNameFilter, Optional<String> workflowTagFilter,
```
Comment (fviale) [inline src/main/java/org/ow2/proactive/catalog/service/CatalogObjectService.java:495]: in general rename occurrences of workflowTag to objectTag

**Comment by fviale** [comment]
LGTM, general comment workflowTag (and all derivatives) should be renamed to objectTag

**Review by fviale — APPROVED** [review]
LGTM

**Review by tao-xinxiu — APPROVED** [review]
LGTM

**Pull request merged**